lib/rrd_updates.ml

```diff
@@ -171,27 +171,36 @@ let of_xml input =
     input
 
 let json_of_t t =
-  let buffer = Buffer.create 4096 in
-
-  let do_data row =
-    Printf.bprintf buffer "{t:%Ld,values:[%s]}" row.time
-      (String.concat "," (List.map Utils.f_to_s (Array.to_list row.row_data)))
+  let open Json in
+  let data_record row =
+    record
+      [
+        ("t", int64 row.time)
+      ; ( "values"
+        , array
+          @@ List.map (fun x -> string (Utils.f_to_s x))
```

Comment:
Why not let Yojson serialize this as float? `Utils.f_to_s` loses precision. Especially for values with a small range this will essentially truncate a lot of values to 0.
If the intention is to limit the size of the json (by not unnecessarily serializing values with more precision than being useful), perhaps using something like `%.4g` instead of `%.4f` would work?

Reply:
Because Yojson fails to produce valid json when NaN and infinities are present. When serializing the values without ~std produces invalid json, with ~std:true it errors out, neither is an option.

I'll use the alternative formatting string